crates/pallet-domains/src/lib.rs

```diff
@@ -194,6 +192,7 @@ mod pallet {
             + AsRef<[u8]>
             + AsMut<[u8]>
             + MaxEncodedLen
+            + Into<Self::Hash>
```

Comment:
What is this doing? Looks suspicious.

Reply:
Good point. Actually this trait bound can be completely eliminated now that I thing about it.